lib/npm.js

```diff
@@ -0,0 +1,132 @@
+import _ from 'lodash';
+import path from 'path';
+import { exec } from 'teen_process';
+import { mkdirp, util } from 'appium-support';
+
+const INSTALL_LOCKFILE = '.appium.install.lock';
+const LINK_LOCKFILE = '.appium.link.lock';
+
+export default class NPM {
+
+  constructor (appiumHome) {
+    this.appiumHome = appiumHome;
+  }
+
+  async exec (cmd, args, opts, execOpts = {}) {
+    let { cwd, json, lockFile } = opts;
+    if (!cwd) {
+      cwd = this.appiumHome;
+    }
+    // ensure the directory we want to install inside of exists
+    await mkdirp(cwd);
+    execOpts.cwd = cwd;
+
+    args.unshift(cmd);
+    if (json) {
+      args.push('-json');
+    }
+    let runner = async () => await exec('npm', args, execOpts);
```

Comment:
The npm commands run by appium will never interface with any other project or user. They'll be run under the hood in a secret directory that is totally managed by Appium. It's therefore up to us whether to use npm or yarn. I picked npm because the user is guaranteed to have it on their system if they're already using Appium.